src/Components/Aspire.Microsoft.Azure.Cosmos/AspireMicrosoftAzureCosmosExtensions.cs

```diff
@@ -33,11 +33,73 @@ public static void AddAzureCosmosClient(
         Action<MicrosoftAzureCosmosSettings>? configureSettings = null,
         Action<CosmosClientOptions>? configureClientOptions = null)
     {
-        AddAzureCosmosClient(builder, configureSettings, configureClientOptions, connectionName, serviceKey: null);
+        ArgumentException.ThrowIfNullOrEmpty(connectionName);
+        var cosmosConnectionInfo = GetCosmosConnectionInfo(builder, connectionName);
+        var settings = builder.GetSettings(connectionName, configureSettings, cosmosConnectionInfo);
+        var clientOptions = builder.GetClientOptions(settings, configureClientOptions);
+        builder.Services.AddSingleton(sp => ConfigureDb(connectionName, settings, clientOptions));
+    }
+
+    /// <summary>
+    /// Registers the <see cref="Database"/> as a singleton in the services provided by the <paramref name="builder"/>.
+    /// </summary>
+    /// <param name="builder">The <see cref="IHostApplicationBuilder" /> to read config from and add services to.</param>
+    /// <param name="connectionName">The connection name to use to find a connection string.</param>
+    /// <param name="configureSettings">An optional method that can be used for customizing the <see cref="MicrosoftAzureCosmosSettings"/>. It's invoked after the settings are read from the configuration.</param>
+    /// <param name="configureClientOptions">An optional method that can be used for customizing the <see cref="CosmosClientOptions"/>.</param>
+    /// <remarks>Reads the configuration from "Aspire:Microsoft:Azure:Cosmos" section.</remarks>
+    /// <exception cref="InvalidOperationException">If required ConnectionString is not provided in configuration section</exception>
+    public static void AddAzureCosmosDatabase(
+        this IHostApplicationBuilder builder,
+        string connectionName,
+        Action<MicrosoftAzureCosmosSettings>? configureSettings = null,
+        Action<CosmosClientOptions>? configureClientOptions = null)
+    {
+        var cosmosConnectionInfo = GetCosmosConnectionInfo(builder, connectionName);
+        if (cosmosConnectionInfo is null || string.IsNullOrEmpty(cosmosConnectionInfo.Value.DatabaseName))
+        {
+            throw new InvalidOperationException($"The connection string '{connectionName}' does not exist or is missing the database name.");
+        }
```

Comment:
Typically we don't throw up front if the connection string isn't passed, but instead wait to see if the information is set in `Action<MicrosoftAzureCosmosSettings>? configureSettings`. Only then do we fail if we couldn't get the info we needed. The connection string initially populates the settings object, and then the app code gets a chance to modify it as well.

Reply:
Delayed the exception to occur when the instance is resolved and updated the settings objects to store the database and/or container name.